src/utils/service_builder.rs

```diff
@@ -398,16 +398,27 @@ fn build_authenticators(config: &AuthenticatorConfig) -> Vec<(AuthType, Authenti
     let mut authenticators: Vec<(AuthType, Authenticator)> = Vec::new();
 
     match config {
-        AuthenticatorConfig::Direct => {
-            authenticators.push((AuthType::Direct, Box::from(DirectAuthenticator {})))
-        }
-        AuthenticatorConfig::UnixPeerCredentials => authenticators.push((
+        AuthenticatorConfig::Direct { admins } => authenticators.push((
+            AuthType::Direct,
+            Box::from(DirectAuthenticator::new(
+                admins.as_ref().cloned().unwrap_or_default(),
```

Comment:
Will the `Default` yield to an empty vector?

Reply:
Yes, that's the default for `Vec`